### ISSUE TITLE
Optuna XGBoost Optmize Study reporting duplicate metric values

### ISSUE THREAD

**Issue by ziadzee** [optuna/optuna#2926]
Hello,

I'm trying to optimise an XGBoost model like so

```python
class Objective:
    
    '''
    The Objective Class is responsible for implementing OpTuna-based hyperparameter tuning.
    '''
    
    def __init__(self):
        self.best_xgboost = None
        self._xgboost = None
        
    def __call__(self, trial: optuna.Trial):
          
        # XGBoost hyperparameters to tune
        params = {
        "verbosity": 1,
        "objective": "binary:logistic",
        "booster" : trial.suggest_categorical("booster", ["gbtree", "dart"]),
        "reg_lambda" : trial.suggest_loguniform("reg_lambda", 1e-8, 1.0),
        "reg_alpha" : trial.suggest_loguniform("reg_alpha", 1e-5, 1.0), 
        "min_child_weight" : trial.suggest_int("min_child_weight", 0, 5),
        "learning_rate" : trial.suggest_int("learning_rate", 1e-05, 1e-1),
        "min_split_loss" : trial.suggest_int("min-split_loss", 0, 5),
        "eval_metric" : trial.suggest_categorical("eval_metric", ["error", "logloss","auc","aucpr"]),
        "n_estimators" : trial.suggest_int("n_estimators", 0, 1000, step = 100)
        }
        
        
        # Initialsing XGBoost model

        model = xgb.XGBClassifier(**params,
                                  random_state = 42)
        
        model.fit(X_train,
                  y_train,
                  early_stopping_rounds = 10,
                  eval_set=[(X_train, y_train), (X_val, y_val)]) 

        
        self.best_xgboost = model

        # Predicting
        prediction = model.predict(X_val)
        prediction_label = np.rint(prediction)
    
        # Evaluating
        f1 = f1_score(y_val, prediction_label)  
        # print("Model Best Score:",model.get_best_score())
    
        return f1
    
    # Callback method that will allow us to save our best performing model once tuning is complete
    def callback(self, study, trial):
        if study.best_trial == trial:
            self.best_xgboost = self._xgboost
```

However, when I run the optimisation study, it returns duplicate values for the metric I'm attempting to optimise like so:

```python
objective = Objective()

# Creating the Optuna study for evaluation with 100 trials

study = optuna.create_study(directions = ["maximize"],
                            sampler = TPESampler(seed=10))

# Optimising study

study.optimize(objective,
               n_trials = 5)

trial = study.best_trials
...
...
```

Output:

```bash
[I 2021-09-15 14:50:27,380] A new study created in memory with name: no-name-4bd5f9f5-bac8-40e7-b323-f7913799147b
[0]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
Multiple eval metrics have been passed: 'validation_1-logloss' will be used for early stopping.

Will train until validation_1-logloss hasn't improved in 10 rounds.
[1]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[2]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[3]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[4]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[5]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[6]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[7]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[8]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[9]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[10]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
Stopping. Best iteration:
[0]	validation_0-logloss:0.692428	validation_1-logloss:0.693246

[I 2021-09-15 14:50:33,833] Trial 0 finished with value: 0.0 and parameters: {'booster': 'gbtree', 'reg_lambda': 0.0011726971762482058, 'reg_alpha': 0.05546505049361988, 'min_child_weight': 2, 'learning_rate': 0, 'min-split_loss': 1, 'eval_metric': 'logloss', 'n_estimators': 700}. Best is trial 0 with value: 0.0.
[0]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
Multiple eval metrics have been passed: 'validation_1-logloss' will be used for early stopping.

Will train until validation_1-logloss hasn't improved in 10 rounds.
[1]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[2]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[3]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[4]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[5]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[6]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[7]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[8]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[9]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
[10]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
Stopping. Best iteration:
[0]	validation_0-logloss:0.692428	validation_1-logloss:0.693246
...
...
...
```

I'm not sure why it's returning exact duplicate values for each trial. I've tried to increase `n_trials` = 100 however the issue persists.

**Comment by himkt** [comment]
Hi @ziadzee, I think  `trial.suggest_int("learning_rate", 1e-05, 1e-1)` would always return `0` (int) for the learning rate, which means it doesn't update weights during training at all. You might have to use `suggest_float` instead.
